sched/group/group_leave.c

```diff
@@ -408,3 +376,66 @@ void group_leave(FAR struct tcb_s *tcb)
 }
 
 #endif /* HAVE_GROUP_MEMBERS */
+
+/****************************************************************************
+ * Name: group_drop
+ *
+ * Description:
+ *   Release the group's memory. This function is called whenever a reference
+ *   to the group structure is released. It is not dependent on member count,
+ *   but rather external references, which include:
+ *   - Waiter list for waitpid()
+ *   - Users of the group's address environment, which can be the group's
+ *     members themselves, as well as kernel tasks whom do not have their own
+ *     address environment.
+ *
+ * Input Parameters:
+ *   group - The group that is to be dropped
+ *
+ * Returned Value:
+ *   None.
+ *
+ * Assumptions:
+ *   Called during task deletion or context switch in a safe context.  No
+ *   special precautions are required here.
+ *
+ ****************************************************************************/
+
+void group_drop(FAR struct task_group_s *group)
+{
+  /* Dummy initial condition */
+
+  if (0)
```

Comment:
let's remove the dummy if:
```
#if defined(CONFIG_SCHED_WAITPID) && !defined(CONFIG_SCHED_HAVE_PARENT)
  if (group->tg_nwaiters > 0)
    {
    }
  else
#endif
#if defined(CONFIG_ARCH_ADDRENV)
  /* If someone needs this address environment, cannot drop it yet */
  if (group->tg_addrenv_refs > 0)
    {
    }
  else
#endif
  /* Finally, if no one needs the group and it has been deleted, remove it */
  if (group->tg_flags & GROUP_FLAG_DELETED)
    {
    }
```

Reply:
Does this actually work, so the compiler knows to combine the else+if in these cases ?

Reply:
why not work? there are many places use this trick.

Reply:
To clarify the thing I did not know is that the compiler knows to parse several lines for the first significant character.

Compare the example you provided to this:
```

  else
#endif
#if defined(CONFIG_ARCH_ADDRENV)
  /* If someone needs this address environment, cannot drop it yet */
  if (group->tg_addrenv_refs > 0)
```

I simply did not realize the compiler knows to ignore the pre-processor directives and the comment line, to combine the else+if. Thus, thank you for the tip!